docs/static/dead-letter-queues.asciidoc

```diff
@@ -0,0 +1,149 @@
+[[dead-letter-queues]]
+=== Dead Letter Queues
+
+NOTE: The dead letter queue feature is currently supported for the
+<<plugins-outputs-elasticsearch>> output only. Support for additional outputs
+will be available in future releases of the Logstash plugins. Before configuring
+Logstash to use this feature, refer to the output plugin documentation to
+verify that the plugin supports the dead letter queue feature.
+
+By default, when Logstash encounters an event that it cannot process because the
+data contains a mapping error or some other issue, the Logstash pipeline 
+either hangs or drops the unsuccessful event. In order to protect against data
+loss in this situation, you can configure Logstash to write unsuccessful events
+to a dead letter queue instead of dropping them. 
+
+Each event written to the dead letter queue includes the original event along
+with metadata indicating when the event entered the queue. For example:
+
+//TODO: Need a better example here. 
+
+[source,json]
+-------------------------------------------------------------------------------
```

Comment:
I confirmed there is no field called `original`. I may have misled @robbavey.

The original event gets appended to the `Event` object at the top-level.

Here is the updated config to deal with mapping error:

```
input { 
  dead_letter_queue { 
    path => "/path.to/data/dead_letter_queue/" 
  } 
} 
filter {
  mutate { 
    remove_field => "[geoip][location]" 
  } 
} 
output { 
  elasticsearch{} 
}
```
This removes the offending field and indexes whatever other field that was present in the original.